apps/consent-api/src/routers/consentReleaseData.ts

```
@@ -0,0 +1,123 @@
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { Router } from 'express';
+import { ConsentReleaseDataRequest } from 'types/entities';
+
+import logger from '../logger.js';
+
+const router = Router();
+
+/**
+ * @openapi
+ * /wizard/steps/consent-to-release-data:
+ *   post:
+ *     tags:
+ *       - Consent Wizard
+ *     name: Submit Consent to Release Data
+ *     description: Form submission for Consent Wizard - Consent to Release Data
+ *     security:
+ *       - bearerAuth: []
+ *     parameters:
+ *       - name: session
+ *         in: header
+ *         required: true
+ *         description: User session token
+ *         schema:
+ *           type: string
+ *     requestBody:
+ *       required: true
+ *       content:
+ *         application/json:
+ *           schema:
+ *             $ref: '#/components/schemas/ConsentReleaseDataRequest'
+ *     responses:
+ *       201:
+ *         description: OK
+ *         content:
+ *           application/json:
+ *             schema:
+ *               $ref: '#/components/schemas/ConsentReleaseDataResponse'
+ *       401:
+ *         description: Unauthorized. Authorization information is missing or invalid.
+ *       403:
+ *         description: Forbidden. Provided Authorization token is valid but has insufficient permissions to make this request.
+ *       500:
+ *         description: Server error
+ */
+router.post('/', async (req, res) => {
+	logger.info(`POST /wizard/steps/consent-to-release-data`);
+	// TODO: implement when auth layer is ready
+	try {
+		logger.info(`Submitted consent to release data`);
+		const data = ConsentReleaseDataRequest.parse(req.body);
```

If this parse fails then an error is thrown, we should be handling this case and communicating to the requester that the request was rejected due to invalid content. This should result in a response with status 400. The response message should ideally include information about what part of the validation failed but translating the zod parsing is a general task that needs focused work so for now just an error 400 would suffice.

We have an opportunity here to write a function to standardize handling POST request body validation, if you want we can talk about building that out quickly.

---

Generally these "map out" tasks have been to lay the groundwork for the api endpoints, but not entirely implement them, including full validation and correct error responses. That is I think where the "build <endpoint>" tickets come in. Currently @mistryrn and I are going through the existing "build" tickets to make sure they're spec'd out correctly - I would love to add in standardized post request handling, maybe we can add as part of these build tickets?

---

I disagree here for two reasons:
1. part of the describing or "mapping" of the endpoint should be getting the error responses correct
2. code was put in place to validate the inputs but the errors were not handled properly

We should get this right as we add in the validation code.